src/util/entropy_utils.js

```diff
@@ -0,0 +1,133 @@
+ *       we used the option onlyStandard for parsing "disk" and "part" entries
+ *       reference: https://github.com/mw-white/node-linux-blockutils
+ * @returns {Promise<object[]>}
+ */
+async function get_block_device_disk_info() {
+    const is_linux = process.platform === 'linux';
+    if (!is_linux) return [];
+
+    const block_devices = await get_block_info_async({ "onlyStandard": true });
+    if (!block_devices) return [];
+
+    const available_disks = [];
+    for (const block_device of block_devices) {
+        if (block_device.TYPE === DEVICE_TYPE_DISK) {
+            available_disks.push({
+                    name: `/dev/${block_device.NAME}`,
+                    size: Number(block_device.SIZE),
+                });
+        }
+    }
+    return available_disks;
+}
+
+/**
+ * pick_a_disk will pick a disk that:
+ *  - The disk name is in the disk_order_by_priority array - the lower index in the array the higher priority
+ *  - Its size is higher than config.ENTROPY_DISK_SIZE_THRESHOLD
+ * 
+ * The item that is returned is an object with properties: name, size and priority
+ * (we could delete the priority property as it is used only inside this function)
+ * 
+ * if none matches then it would return undefined
+ * @param {object[]} available_disks
+ * @returns {object | undefined}
+ */
+function pick_a_disk(available_disks) {
+    // the disk_order_by_priority array is the whitelist and also sets the priority
+    // (0 index highest priority, n-1 index lowest priority)
+    // note: we decided on the order according to previous implantation we had in the past
+    // as it added in patches we might want to reconsider it
+    const disk_order_by_priority = ['/dev/sd', '/dev/vd', '/dev/xvd', 'dev/dasd', '/dev/nvme'];
+    const priority_disk_array = [];
+    for (const disk of available_disks) {
+        if (disk.size > config.ENTROPY_DISK_SIZE_THRESHOLD) {
+            for (let index = 0; index < disk_order_by_priority.length; index++) {
+                const prefix = disk_order_by_priority[index];
+                if (disk.name.startsWith(prefix)) {
+                    disk.priority = index;
+                    priority_disk_array.push(disk);
+                }
+            }
+        }
+    }
+    // sort the array based on the priority
+    const sorted_priority_disk_array = priority_disk_array.sort((item1, item2) => item1.priority - item2.priority);
+    const first_element = sorted_priority_disk_array.length > 0 ? sorted_priority_disk_array[0] : undefined;
+    return first_element;
+}
+
+/**
+ * add_entropy will execute dd command which pipes to md5sum
+ * in order to get the MD5 hash to change
+ * @param {string} disk_name
+ * @param {number} disk_size
+ */
+async function add_entropy(disk_name, disk_size) {
+    const bs = 1024 * 1024;
+    const count = 32;
+    const disk_size_in_blocks = disk_size / bs;
+    const skip = chance.integer({ min: 0, max: disk_size_in_blocks });
```

Comment:
I think I originally forgot to reduce 1 from the max because I see that the range is inclusive in chance.integer()
```suggestion
    const skip = chance.integer({ min: 0, max: disk_size_in_blocks - 1 });
```